linchpin/api/__init__.py

```diff
@@ -407,7 +407,13 @@ def setup_rundb(self):
             rundb_conn_dir = os.path.dirname(rundb_conn_int)
 
             if not os.path.exists(rundb_conn_dir):
-                os.mkdir(rundb_conn_dir)
+                try:
+                    os.makedirs(rundb_conn_dir)
+                except OSError as exc:
+                    if exc.errno == errno.EEXIST and os.path.isdir(path):
+                        pass
+                    else:
+                        raise
```

Comment:
@herlo Does this raise a empty exception ?? without any message ?? 
A message would to raise would be better . 
```
raise Exception('some relevant message ') 
```

Reply:
@samvarankashyap https://stackoverflow.com/questions/18001721/raise-with-no-argument - Essentially, it reraises the last known exception in the stack. It's fairly standard practice I've seen in the past. it does look like I need to fix some small issues though.